### PR TITLE
Move internal index templates to composable templates

### PR DESCRIPTION
**Pull request by probakowski**
This change moves watcher, ILM history and SLM history templates to composable templates.
Versions are updated to reflect the switch. Only change to the templates themselves is added `_meta` to mark them as managed

### CODE REVIEW

**Comment by probakowski** [comment]
@elasticmachine run elasticsearch-ci/bwc 
run elasticsearch-ci/default-distro

**Comment by probakowski** [comment]
@elasticmachine update branch

**Review by pugnascotia — APPROVED** [review]
LGTM but you shouldn't take my word for it 😅

@dakrone?

File: x-pack/plugin/core/src/main/resources/triggered-watches.json [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-    "index.number_of_shards": 1,
-    "index.auto_expand_replicas": "0-1",
-    "index.refresh_interval" : "-1",
-    "index.format": 6,
-    "index.priority": 900
-  },
-  "mappings": {
-    "_doc": {
-      "dynamic" : "strict",
+  "priority": 2147483647,
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/triggered-watches.json:3]: Can you add the `_meta` map to this template as well?

File: x-pack/plugin/core/src/main/resources/watch-history-no-ilm.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-  "settings": {
-    "index.number_of_shards": 1,
-    "index.number_of_replicas": 0,
-    "index.auto_expand_replicas": "0-1",
-    "index.hidden": true,
-    "index.format": 6
-  },
-  "mappings": {
-    "doc": {
+  "priority": 2147483646,
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/watch-history-no-ilm.json:3]: This template also needs a `_meta` map

File: x-pack/plugin/core/src/main/resources/watch-history.json [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-    "index.number_of_shards": 1,
-    "index.number_of_replicas": 0,
-    "index.auto_expand_replicas": "0-1",
-    "index.lifecycle.name": "watch-history-ilm-policy",
-    "index.hidden": true,
-    "index.format": 6
-  },
-  "mappings": {
-    "_doc": {
+  "priority": 2147483647,
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/watch-history.json:3]: Needs a `_meta` map

File: x-pack/plugin/core/src/main/resources/watches.json [hunk 4]
```diff
@@ -1,62 +1,62 @@
...
-    "index.number_of_shards": 1,
-    "index.number_of_replicas": 0,
-    "index.auto_expand_replicas": "0-1",
-    "index.format": 6,
-    "index.priority": 800
-  },
-  "mappings": {
-    "_doc": {
-      "dynamic" : "strict",
+  "priority": 2147483647,
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/watches.json:3]: Same about needing a `_meta` map

**Review by dakrone — CHANGES_REQUESTED** [review]
Thanks for working on this @probakowski, I left some comments about missing `_meta` maps in some of the templates (we need these to indicate in the UI that a template is managed by ES automatically).

I was thinking though, that we should increment the version to indicate the change, that way a user is not left with a `version: 2` legacy template and `version: 2` composable index template; they can immediately tell from the `version: 3` signifier that the composable index template version is "newer" (since the old one will not be removed) even if they are not familiar with the difference between the types of templates.

File: x-pack/plugin/core/src/main/resources/triggered-watches.json (hunk 1 above)
Comment (probakowski) [reply x-pack/plugin/core/src/main/resources/triggered-watches.json:3]: You're right, I forgot about `_meta` again. I've added it in all templates

**Comment by probakowski** [comment]
@dakrone that makes sense, I've increased version in all templates

**Review by dakrone — APPROVED** [review]
LGTM, but there is one more missing version increment, in `ILMHistoryTemplateRegistry` we need to increment the version for the ilm history index template

**Comment by probakowski** [comment]
Good catch, I've updated it as well, thanks!

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryTemplateRegistry.java [hunk 5]
```diff
@@ -28,7 +28,8 @@
     // history (please add a comment why you increased the version here)
     // version 1: initial
     // version 2: convert to hidden index
-    public static final int INDEX_TEMPLATE_VERSION = 2;
+    // version 2: templates moved to composable templates
```
Comment (dakrone) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryTemplateRegistry.java:31]: Super minor typo
```suggestion
    // version 3: templates moved to composable templates
```

**Comment by probakowski** [comment]
pinging @elastic/es-ui, FYI this change updates version field in `WatcherIndexTemplateRegistryField` so probably something around watcher ui will need to be updated as well.
Also versions in SLM and ILM are updated, not sure if this is relevant for UI

**Comment by cjcenizal** [comment]
Thank you for the ping @probakowski! To help me understand what needs to change in the UI, could you please describe how the watcher, SLM, and ILM APIs have changed?

**Comment by probakowski** [comment]
@cjcenizal I don't know if you consider this API but versions templates and some internal indices got upgraded:

- ILM history index was `ilm-history-2`, now it will be `ilm-history-3`
- SLM history index was `.slm-history-2`, now it will be `.slm-history-3`
- Watcher history index was `.watcher-history-11`, now it will be `.watcher-history-12`

According to note in [WatcherIndexTemplateRegistryField](https://github.com/probakowski/elasticsearch/blob/dacba2e5b1fbfb7bb3441bb614f0dc2ebdbdce11/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/WatcherIndexTemplateRegistryField.java#L20) UI should be notified when these version changes as UI needs to be updated

**Pull request merged**